server/src/main/java/com/cloud/acl/DomainChecker.java

```diff
@@ -167,81 +182,155 @@ public boolean checkAccess(User user, ControlledEntity entity) throws Permission
     }
 
     @Override
-    public boolean checkAccess(Account account, DiskOffering dof) throws PermissionDeniedException {
-        if (account == null || dof == null || dof.getDomainId() == null) {//public offering
-            return true;
+    public boolean checkAccess(Account account, DiskOffering dof, DataCenter zone) throws PermissionDeniedException {
+        boolean isAccess = false;
```

Comment:
minor nit: `hasAccess` is perhaps a better variable name :)

Reply:
Done